client/src/styles/legacy/GridField.scss

```diff
@@ -424,3 +424,7 @@
     padding: $grid-gutter-width;
   }
 }
+
+#Form_ItemEditForm #preview-mode-dropdown-in-content {
```

Comment:
There was some specific combination of things that this was doing - if I can remember what it was I'll add a comment and see if I can figure out a nicer selector for it (I agree double ID is trash, I just couldn't figure a cleaner way to do it at the time).
If I can't remember what it was I'll remove it out-right.

Reply:
AHA! I remember what this is for now. `margin-left: auto` pushes GridField buttons off to the left, with an unnecessarily large gap:
![image](https://user-images.githubusercontent.com/36352093/151287675-756b99f6-8332-4232-9654-0c90720e2d08.png)
Whereas `margin-left: 0` doesn't add all that extra space - the preview mode selector just sits nicely next to the GridField buttons.
![image](https://user-images.githubusercontent.com/36352093/151287726-96bdec5a-f320-423c-99b1-1056e7d04c73.png)

Reply:
The double ID selector is because I need to grab that specific element but _only_ if it's within a GridFieldDetailForm.... and I'm still not quite sure how else to do that.

Reply:
I've thought about it and I haven't been able to think of another way get a selector that does what we need here. If anyone can think of one let me know but for now I'm just going to add a comment explaining why it is there and what it does.
I'll also add it to _preview.scss instead of GridField.scss.... I must not have noticed that file the first time around.

Reply:
Ah OK.  I'm not actually sure how to get that group of buttons (left, right, green plus).  What module(s) are you using to get those to show?

Was that group of button previously right aligned?  If so then I think it's probably OK to include this new rule.  If they were previously left aligned though then we should keep `margin: auto;`

Reply:
They're part of Framework. For context, this is inside the gridfield item edit form for a `DataObject` inside a custom `ModelAdmin` - though there's nothing special happening in the admin to add or alter components in the gridfield. They should display for any `GridField` that has a `GridFieldDetailForm` with `showAdd` and `showPagination` both set to true.
They were already right aligned.

Reply:
Yup OK I can see why we need this rule now:

I've left setup notes to replicate as a comment below in this PR

**With:**

![image](https://user-images.githubusercontent.com/4809037/151455931-cbb3ec19-2112-4fe0-9a18-001fc7acb58c.png)


**Without:**

![image](https://user-images.githubusercontent.com/4809037/151455976-21df0ef4-96e6-47d0-880a-50c66eacb92e.png)

Reply:
For the sake of anyone who is wondering why the `margin-left: auto` rule is there in the first place, it's for the original use case for previews which is in the site tree edit form. Without it, the preview selector is no longer right aligned:
![image](https://user-images.githubusercontent.com/36352093/151456510-99d859cd-a4c7-48a1-adb0-2c3af31a2049.png)

Reply:
Just to tidy up, could you change it to this

**client/src/styles/legacy/_preview.scss:259**
```
.toolbar--south > .btn-toolbar > .preview-selector {
  margin-left: auto;

  &#preview-mode-dropdown-in-content {
    margin-left: 0;
  }
}
```

I think that's probably as tidy as we'll get this without risking a css regression somewhere else

Reply:
That would also select the SiteTree edit form's previewmode dropdown, which would result in making that one left aligned.